velox/expression/CastExpr.cpp

```diff
@@ -59,6 +59,17 @@ void applyCastKernel(
     auto writer = exec::StringWriter<>(result, row);
     writer.copy_from(output);
     writer.finalize();
+  } else if (isDecimalType(*input->type())) {
+    // Special handling for decimal types.
+    auto [precision, scale] = getDecimalPrecisionScale(*input->type());
+    if constexpr (ToKind == TypeKind::DOUBLE) {
+      auto output =
+          util::Converter<CppToType<double>::typeKind, void, Truncate>::cast(
+              input->valueAt(row), nullOutput);
+      if (!nullOutput) {
+        result->set(row, output / (double)DecimalUtil::kPowersOfTen[scale]);
```

Comment:
Does this satisfy the divide half round up?

Reply:
I don't think rounding is an issue here